### PR TITLE
Harden repository contract tests

### PR DESCRIPTION
**Pull request by kmozze**
## Что сделано

- Добавлен Database Rider для repository integration tests: тесты теперь задают состояние БД через fixtures и проверяют итоговое состояние там, где это важно.
- CategoryRepositoryTest переведен на dataset-based стиль: отдельно проверяется создание новой категории и идемпотентность createIfAbsent для дубля имени пользователя.
- ExpenseRepositoryTest укреплен для контракта findAllByUserIdAndPeriod: from включается, to исключается, чужой user_id не попадает, пустой период возвращает emptyList(), сортировка newest-first сохраняется (замечание из PR #24: пункт про repository contract tests для findAllByUserIdAndPeriod).
- Ручные setup helpers и прямой DSL update created_at удалены из ExpenseRepositoryTest; точные TIMESTAMP WITH TIME ZONE boundary values сидятся SQL fixture.

## Как проверить

- Автоматические проверки пройдены.

## Ревьюеры

- Danil Mozzhevilov (`Dmozze`)

### CODE REVIEW

File: src/test/resources/datasets/repository/expense-period.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        40001,
+        timestamptz '2026-05-21T00:00:00Z'
+    ),
+    (
+        '00000000-0000-0000-0000-000000000106',
+        600.00,
+        '00000000-0000-0000-0000-000000000002',
+        40002,
+        timestamptz '2026-05-20T13:00:00Z'
+    );
```
Comment (Dmozze) [inline src/test/resources/datasets/repository/expense-period.sql:53]: 👍 Эта fixture — **6 записей, фиксирующих 4 invariant'а одним тестом**:

| Запись | Invariant |
| --- | --- |
| `101` createdAt = `2026-05-20T00:00:00Z` (= from) | **from-inclusive** |
| `102` createdAt = `2026-05-20T10:00:00Z` (внутри) | ordering check |
| `103` createdAt = `2026-05-20T12:00:00Z` (внутри) | ordering check |
| `104` createdAt = `2026-05-19T23:59:59Z` (< from) | **excluded before** |
| `105` createdAt = `2026-05-21T00:00:00Z` (= to) | **to-exclusive** |
| `106` чужой user_id = 40002 | **user isolation** |

Ожидаемый порядок: `103, 102, 101` (newest → oldest, без 104/105/106).

Это **не "6 тестов в одном"**, это **один тест, проверяющий 4 свойства контракта одной выборкой**. Очень дешёво в поддержке (один fixture файл), очень дорого в покрытии. Архитектурно правильно.

Отдельный плюс: invariants **читаются прямо из fixture** — ревьюер видит boundary values в самом datasource, не приходится распутывать через сетапы в Kotlin коде.

File: src/test/kotlin/me/kmozze/expensetracker/integration/repository/CategoryRepositoryTest.kt [hunk 2]
```diff
@@ -1,42 +1,78 @@
...
+    caseInsensitiveStrategy = Orthography.LOWERCASE,
+    dataTypeFactoryClass = PostgresqlDataTypeFactory::class,
+)
 class CategoryRepositoryTest : AbstractIntegrationTest() {
     @Autowired
     private lateinit var categoryRepository: ICategoryRepository
 
     @Test
-    fun `create if absent ignores duplicate category name for user`() {
+    @DataSet(
```
Comment (Dmozze) [inline src/test/kotlin/me/kmozze/expensetracker/integration/repository/CategoryRepositoryTest.kt:27]: Двухстадийный fixture pattern: `category-empty.yml` для setup → `category-created.yml` через `@ExpectedDataSet` для проверки.

`ignoreCols = ["created_at", "updated_at"]` — правильный хвост: timestamps выставляются БД (`DEFAULT CURRENT_TIMESTAMP`), их сравнивать смысла нет. Это **invariant**, который видно прямо в декларации теста, без условных ассертов в Kotlin.

**Review by Dmozze — APPROVED** [review]
LGTM. Закрывает мой [☑️ из PR #24](https://github.com/kmozze/expense-tracker-bot/pull/24#discussion_r...) (boundary-тесты для `findAllByUserIdAndPeriod`) **сразу 4 invariant'а одним тестом** через продуманную SQL fixture. Уровень decoupling test data ↔ test logic — выше, чем я ожидал.

**Архитектурно важно:**

Database Rider — это переход от "insertion-based test" (вставляю записи в setUp, проверяю выборку) к **"declarative test data"** (fixture в YAML/SQL, тест декларативно описывает что должно остаться). Преимущества:

1. **`TIMESTAMP WITH TIME ZONE` boundary values** теперь хранятся в SQL как литералы (`timestamptz '2026-05-20T00:00:00Z'`). Раньше тест писал значения через `dsl.update(CREATED_AT, ...)` — а сам этот update мог округлять/конвертить таймстемп. Теперь boundary точные, без шанса round-trip искажения.
2. **`dsl.update(CREATED_AT)` хак удалён** — это был [мой 🟡 #5 в PR #24 review](https://github.com/kmozze/expense-tracker-bot/pull/24#discussion_r...). Закрыт через структурное решение, не через workaround.
3. **Read-path и write-path тестируются отдельно.** `CategoryRepositoryTest.create_if_absent_creates_category_when_missing` — это **only write** (`@ExpectedDataSet` проверяет конечное состояние БД). `findAllByUserIdAndPeriod` — это **only read** (fixture фиксирует начальное состояние).

Approve.
